src/Features/Core/Portable/ConvertTupleToStruct/AbstractConvertTupleToStructCodeRefactoringProvider.cs

```diff
@@ -163,7 +156,59 @@ private static string GetTitle(Scope scope)
             return (expressionOrType, tupleType);
         }
 
-        private async Task<Solution> ConvertToStructAsync(
+        public async Task<Solution> ConvertToStructAsync(
+            Document document, TextSpan span, Scope scope, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            using (Logger.LogBlock(FunctionId.AbstractConvertTupleToStructCodeRefactoringProvider_ConvertToStructAsync, cancellationToken))
+            {
+                var solution = document.Project.Solution;
+                var client = await RemoteHostClient.TryGetClientAsync(solution.Workspace, cancellationToken).ConfigureAwait(false);
+                if (client != null)
+                {
+                    var resultOpt = await client.TryRunRemoteAsync<SerializableConvertTupleToStructResult>(
+                        WellKnownServiceHubServices.CodeAnalysisService,
+                        nameof(IRemoteConvertTupleToStructCodeRefactoringProvider.ConvertToStructAsync),
+                        solution,
+                        new object[]
+                        {
+                            document.Id,
+                            span,
+                            scope,
+                        },
+                        callbackTarget: null,
+                        cancellationToken).ConfigureAwait(false);
+
+                    if (resultOpt.HasValue)
+                    {
+                        var result = resultOpt.Value;
+                        var resultSolution = await RemoteUtilities.UpdateSolutionAsync(
+                            solution, result.DocumentTextChanges, cancellationToken).ConfigureAwait(false);
+                        return await AddRenameTokenAsync(
+                            resultSolution, result.RenamedToken, cancellationToken).ConfigureAwait(false);
+                    }
+                }
+            }
+
+            return await ConvertToStructInCurrentProcessAsync(
+                document, span, scope, cancellationToken).ConfigureAwait(false);
+        }
+
+        private async Task<Solution> AddRenameTokenAsync(
```

Comment:
Hmm, so annotations get lost during serialization? Is it why we need to keep track of them here?

Reply:
Yes.  We'd like to address that in a later change, but it will be a significant amount of work to support.